### PR TITLE
Promises

### PR DESCRIPTION
**Pull request by dlongley**
Do not merge until #31 is merged.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,11 +3,11 @@
...
 A MongoDB ledger storage subsystem for bedrock-ledger that enables the
-storage and retrieval of ledgers, blocks, and events. The relationship
-of these objects are shown below:
+storage and retrieval of ledgers, blocks, events, and operations. The
+relationship of these objects are shown below:
 
-<img alt="Ledgers contain blocks, blocks contain events"
-  src="https://w3c.github.io/web-ledger/diagrams/blocks.svg"
+<img alt= 'Ledgers contain blocks, blocks contain events '
+  src= 'https://w3c.github.io/web-ledger/diagrams/blocks.svg '
```
Comment (davidlehn) [inline README.md]: Looks like over zealous quote search & replace.

Comment (dlongley) [reply README.md]: It's probably in the multihash branch too... I'll just fix it here.

**Comment by mattcollier** [comment]
Here's the output of the performance tests for master / mutihash / promises

https://gist.github.com/mattcollier/53753d7932b4b35c176fe6466ea031a6

File: lib/index.js [hunk 2]
```diff
@@ -62,412 +60,316 @@ bedrock.events.on('bedrock.start', () => {
...
-        fields: {'recordId': 1},
-        options: {unique: false, background: false, name: 'operationIndex2'}
-      }], callback);
-    }],
-    // storage plugins add indexes to collections by providing an
-    // `expandIndexes` method
-    expandIndexes: ['createCollectionIndexes', (results, callback) => {
-      if(plugins.length === 0) {
-        return callback();
+  const insertedRecord = (await database.collections.ledger.insert(
```
Comment (mattcollier) [inline lib/index.js]: are there plans for this unused var?

File: lib/LedgerBlockStorage.js [hunk 3]
```diff
@@ -291,221 +251,164 @@ class LedgerBlockStorage {
...
+    if(!record) {
+      throw new BedrockError(
+        'A block with the given block height does not exist.',
+        'NotFoundError', {blockHeight});
+    }
+    if(eventHash) {
+      // TODO: make code DRY
+      // FIXME: this might be accomplished with aggregate query
+      // get event hashes
+      const block = results.find.block;
```
Comment (mattcollier) [inline lib/LedgerBlockStorage.js]: `results` is undefined, is this supposed to be `record.block`?

File: lib/LedgerBlockStorage.js [hunk 4]
```diff
@@ -185,104 +174,75 @@ class LedgerBlockStorage {
...
-          'block.blockHeight': blockHeight,
-          'meta.deleted': {
-            $exists: false
-          },
-          'meta.consensus': {
-            $exists: true
-          }
-        };
-        this.collection.findOne(query, callback);
+  async getByHeight(blockHeight) {
```
Comment (mattcollier) [inline lib/LedgerBlockStorage.js]: I found it curious that the error below was not detected and it appears to me that this API is not used anywhere.  Does it seem useful?  Should it be dropped?

Comment (dlongley) [reply lib/LedgerBlockStorage.js]: It should either be tested or dropped IMO ... but not in this PR.

**Comment by dlongley** [comment]
Ready for another pass.

**Pull request merged**